cmd/README.md

```diff
@@ -0,0 +1,521 @@
+pg_dump/mysqldump output. For example, suppose you want to try out different 
+primary keys for a table. First run pg_dump/mysqldump and save the output to 
+a file. Then modify (or add) the relevant 
+`ALTER TABLE ... ADD CONSTRAINT ... PRIMARY KEY ...` statement in the 
+pg_dump/mysqldump output file so that the primary keys match what you need. 
+Then run HarbourBridge on the modified pg_dump/mysqldump output.
+
+## Files Generated by HarbourBridge
+
+HarbourBridge generates several files as it runs:
+
+- Schema file (ending in `schema.txt`): contains the generated Spanner
+  schema, interspersed with comments that cross-reference to the relevant
+  PostgreSQL/MySQL schema definitions.
+
+- Session file (ending in `session.json`): contains all schema and data
+  conversion state endcoded as JSON. It is basically a snapshot of the session.
+
+- Report file (ending in `report.txt`): contains a detailed analysis of the
+  PostgreSQL/MySQL to Spanner migration, including table-by-table stats and an
+  analysis of PostgreSQL/MySQL types that don't cleanly map onto Spanner types. 
+  Note that PostgreSQL/MySQL types that don't have a corresponding Spanner type 
+  are mapped to STRING(MAX).
+
+- Bad data file (ending in `dropped.txt`): contains details of data
+  that could not be converted and written to Spanner, including sample
+  bad-data rows. If there is no bad-data, this file is not written (and we
+  delete any existing file with the same name from a previous run).
+
+By default, these files are prefixed by the name of the Spanner database (with a
+dot separator). The file prefix can be overridden using the `-prefix`
+[option](#options).
+
+## Command line flags
+
+`-source` Required flag. Specifies the source database. Supported source
+databases are _'postgres'_, _'mysql'_ and _'dynamodb'_.
+
+`-target` Optional flag. Specifies the target database. Defaults to _'spanner'_ 
+, which is the only supported target database today.
+
+`-prefix` Specifies a file prefix for the report, schema, and bad-data files
+written by the tool. If no file prefix is specified, the name of the Spanner
+database (plus a '.') is used.
+
+`-v` or `-verbose` Specifies verbose mode. This will cause HarbourBridge to
+output detailed messages about the conversion.
+
+`-skip-foreign-keys` Controls whether we add foreign key constraints after
+data migration is complete. This flag cannot be used with schema-only mode,
+and does not affect the generation of foreign key statements during schema
+processing i.e. foreign key constraints will still appear in the generated 
+Spanner DDL files.
+
+`-session` Specifies a session file that contains all schema and data 
+conversion state endcoded as JSON.
+
+### Source profile (`-source-profile`)
+
+HarbourBridge accepts the following options for --source-profile,
+specified as "key1=value1,key2=value,..." pairs:
+
+`-schema-sample-size` Specifies the number of rows to use for inferring schema 
```

Comment:
suggest we move this to the dynamodb specific readme. We should keep only generic stuff here. I think it makes sense to move any DB specific stuff into the source profile, so this flag would go there

Reply:
Done.